cmd/minikube/cmd/start.go

```diff
@@ -402,29 +401,31 @@ func provisionWithDriver(cmd *cobra.Command, ds registry.DriverState, existing *
 	}, nil
 }
 
-func virtualBoxMacOS13PlusWarning(driverName string) {
-	if !driver.IsVirtualBox(driverName) || !detect.MacOS13Plus() {
+// warnDriverDeprecated prints a small deprecation warning if the selected driver is deprecated
+func warnDriverDeprecated(driverName string) {
+	if driverName == "" {
 		return
 	}
-	out.WarningT(`Due to changes in macOS 13+ minikube doesn't currently support VirtualBox. You can use alternative drivers such as 'vfkit', 'qemu', or 'docker'.
-    https://minikube.sigs.k8s.io/docs/drivers/vfkit/
-    https://minikube.sigs.k8s.io/docs/drivers/qemu/
-    https://minikube.sigs.k8s.io/docs/drivers/docker/
-    For more details on the issue see: https://github.com/kubernetes/minikube/issues/15274
-`)
-}
 
-// hyperkitDeprecationWarning prints a deprecation warning for the hyperkit driver
-func hyperkitDeprecationWarning(driverName string) {
-	if !driver.IsHyperKit(driverName) {
-		return
+	for _, d := range registry.List() {
```

Comment:
Looping over all drivers is not needed, since we have the driver name. I think we better get the driver from the registry and check if the driver is deprecated.

Can we done using:

```
d := registry.Driver(driverName)
if d.Name == "" {
    // fail, no such driver
}

If d.Priority == registry.Deprecated {
    // show warning
}
```